database/sqlite/sqlite_functions.c

```diff
@@ -2208,3 +2212,263 @@ void sql_load_node_id(RRDHOST *host)
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to fetch gaps");
+        return SQLITE_ERROR;
+    };
+
+    rc = sqlite3_bind_text(res, 1, host->machine_guid, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind host_id parameter to load gap information");
+        goto failed;
+    }
+
+    do {
+        rc = sqlite3_step(res);
+        switch (rc) {
+            case SQLITE_ROW:
+                if (likely(sqlite3_column_bytes(res, 0) == sizeof(uuid_t))) {
+                    set_host_gap(host, res);
+                    debug(D_REPLICATION, "%s: Setting %s host latest gap completed!", REPLICATION_MSG, host->hostname);
+                }
+                break;
+            case SQLITE_DONE:
+                set_host_gap(host, NULL);
+                debug(D_REPLICATION, "%s: Setting %s host gaps from SQLite completed with NO ROWs!", REPLICATION_MSG, host->hostname);
+                break;
+            default:
+                debug(D_REPLICATION, "%s: Setting %s host gaps from SQLite returned unexpected error code!", REPLICATION_MSG, host->hostname);
+                break;
+        }
+    } while (rc == SQLITE_ROW);
+
+failed:
+    if (unlikely(sqlite3_finalize(res) != SQLITE_OK))
+        error_report("Failed to finalize the prepared statement when loading all gaps information");
+    return rc;
+};
+
+/*
+ * Delete all gaps from the database
+ */
+int sql_delete_all_gaps(void)
+{
+    sqlite3_stmt *res = NULL;
+    int rc;
+
+    if (unlikely(!res)) {
+        rc = prepare_statement(db_meta, DELETE_ALL_GAPS, &res);
+        if (rc != SQLITE_OK) {
+            error_report("Failed to prepare statement to delete all gaps");
+            return rc;
+        }
+    }
+
+    rc = sqlite3_step(res);
+    if (unlikely(rc != SQLITE_DONE))
+        error_report("Failed to delete gaps, rc = %d", rc);
+
+    rc = sqlite3_reset(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to reset statement when deleting gaps, rc = %d", rc);
+    return rc;
+}
+
+/*
+ * Delete all the gaps of a host from the metadata DB.
+ */
+int sql_delete_all_host_gaps(RRDHOST *host)
+{
+    sqlite3_stmt *res = NULL;
+    int rc;
+
+#ifdef NETDATA_INTERNAL_CHECKS
+    debug(D_METADATALOG,"Deleting all gaps from host %s", host->hostname);
+#endif
+
+    if (unlikely(!res)) {
+        rc = prepare_statement(db_meta, DELETE_ALL_HOST_GAPS, &res);
+        if (rc != SQLITE_OK) {
+            error_report("Failed to prepare statement to delete all gaps for host %s", host->hostname);
+            return rc;
+        }
+    }
+
+    rc = sqlite3_bind_text(res, 1, host->machine_guid, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind MGUID[%s] parameter to delete all gaps for host %s.", host->machine_guid, host->hostname);
+        goto bind_fail;
+    }     
+
+    rc = sqlite3_step(res);
```

Comment:
Please, see [this comment](https://github.com/netdata/netdata/pull/12556/files#r879026869)